### DISCUSSION TITLE
Add custom 'section headers' to ui.table

### DISCUSSION THREAD

**Discussion by snowbollaanm** [zauberzeug/nicegui#2118] [Q&A] [answered]
### Question

Hi! 

I'm trying to give a large dataset table 3 'headers' for each section in a table. Like this attached Image:
![image](https://github.com/zauberzeug/nicegui/assets/139186074/420faa0b-cf55-4dfa-9109-b35ba691071c)

I would use three separate tables however these columns will be sortable, and I need to compare each section.

I've tried using 'add-slot' to no avail so far and was hoping someone could help! 

```
            runTable.add_slot('top', r'''
                <thead>
                <tr>
                    <th colspan="3" style="text-align:center;">Day</th>
                    <th colspan="3" style="text-align:center;">Week</th>
                    <th colspan="3" style="text-align:center;">Month</th>
                </tr>
                </thead>
                ''')
```

Thanks

**Comment by falkoschindler** [comment, marked as answer]
The "top" slot seems to live outside of the HTML table, so tr and th tags are ineffective.
But you can overwrite the "header" slot like this:

```py
columns = [
    {'name': 'a', 'label': 'A', 'field': 'a'},
    {'name': 'b', 'label': 'B', 'field': 'b'},
    {'name': 'c', 'label': 'C', 'field': 'c'},
    {'name': 'd', 'label': 'D', 'field': 'd'},
]
rows = [
    {'a': 1, 'b': 2, 'c': 3, 'd': 4},
    {'a': 5, 'b': 6, 'c': 7, 'd': 8},
]
table = ui.table(columns=columns, rows=rows, row_key='a')
table.add_slot('header', '''
    <tr>
        <th colspan="2">A&B</th>
        <th colspan="2">C&D</th>
    </tr>
    <tr :props="props">
        <th v-for="col in props.cols" :key="col.name">
            {{ col.label }}
        </th>
    </tr>
''')
```
<img width="179" alt="Screenshot 2023-12-04 at 07 28 33" src="https://github.com/zauberzeug/nicegui/assets/5767091/cc285011-11c2-4460-adb2-b4d3e113133d">

**Reply by snowbollaanm** [reply]
I've just discovered an issue that when setting this slot as a rewrites over the Quasar 'sortable' and 'align' functionality that they no longer work. Do you know a fix to this?

**Reply by falkoschindler** [reply]
No, unfortunately I don't. Maybe that's a question for StackOverflow or some Quasar-related forum, since the problem independent of NiceGUI.
